snuba/datasets/dataset_schemas.py

```diff
@@ -24,21 +25,38 @@ def __init__(
     def get_read_schema(self) -> Schema:
         return self.__read_schema
 
-    def get_write_schema(self) -> Schema:
+    def get_write_schema(self) -> Union[WritableTableSchema, None]:
         return self.__write_schema
 
-    def __get_unique_schemas(self) -> List[Schema]:
+    def get_write_schema_enforce(self) -> WritableTableSchema:
```

Comment:
Why is this a separate method, instead of an option on `get_write_schema`? Is this just to make type checking a bit easier? Maybe it would make sense to split the concept read-only and read-write datasets at some point? (You probably have already thought of this.)

Reply:
It is for typing, at least we can designate who needs a valid writer and tell it apart from who may or may not need it.
When we will split dataset from tablestorage this should not be a problem anymore.
I thought about splitting the dataset in this PR originally, It just kept seeming orthogonal to this diff so I reduced the impact.